sdk/src/main/java/com/uid2/sdk/UID2Manager.kt

```diff
@@ -0,0 +1,242 @@
+package com.uid2.sdk
+
+import android.content.Context
+import com.uid2.sdk.UID2ManagerState.IdentityStatusChange
+import com.uid2.sdk.data.IdentityPackage
+import com.uid2.sdk.data.IdentityStatus
+import com.uid2.sdk.data.IdentityStatus.ESTABLISHED
+import com.uid2.sdk.data.IdentityStatus.EXPIRED
+import com.uid2.sdk.data.IdentityStatus.INVALID
+import com.uid2.sdk.data.IdentityStatus.NO_IDENTITY
+import com.uid2.sdk.data.IdentityStatus.OPT_OUT
+import com.uid2.sdk.data.IdentityStatus.REFRESHED
+import com.uid2.sdk.data.IdentityStatus.REFRESH_EXPIRED
+import com.uid2.sdk.data.UID2Identity
+import com.uid2.sdk.extensions.getMetadata
+import com.uid2.sdk.network.DefaultNetworkSession
+import com.uid2.sdk.network.NetworkSession
+import com.uid2.sdk.storage.StorageManager
+import com.uid2.sdk.utils.TimeUtils
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+
+/**
+ * A listener interface allowing the consumer to be notified when either the identity or status of the identity changes
+ * within the UID2Manager.
+ */
+interface UID2ManagerIdentityChangedListener {
+    fun onIdentityStatusChanged(identity: UID2Identity?, status: IdentityStatus)
```

Comment:
This is a good way to publish events.  Keeps the API and dependencies light weight.  👍

Reply:
This is intended for usage by Java, so cannot be `sealed`